homeassistant/components/minecraft_server/config_flow.py

```diff
@@ -52,9 +57,8 @@ async def async_step_user(self, user_input=None) -> FlowResult:
                     CONF_HOST: host,
                     CONF_PORT: port,
                 }
-                server = MinecraftServer(self.hass, "dummy_unique_id", config_data)
-                await server.async_check_connection()
-                if not server.online:
+                server_online = await self.async_is_server_online(config_data)
+                if not server_online:
                     # Host or port invalid or server not reachable.
                     errors["base"] = "cannot_connect"
                 else:
```

Comment:
```suggestion
                if (await self._async_is_server_online(host, port)):
                    # Configuration data are available and no error was detected,
                    # create configuration entry.
                    return self.async_create_entry(title=title, data=config_data)
                else:
                    # Host or port invalid or server not reachable.
                    errors["base"] = "cannot_connect"
```

This would be better for the readability.
(Cannot select all lines for the suggestion)